iotedgedev/deploymentmanifest.py

```diff
@@ -146,7 +146,7 @@ def validate_deployment_template(self):
         validation_success = True
         try:
             template_schema = json.loads(urlopen(Constants.deployment_template_schema_url).read().decode())
-            self._validate_json_schema(template_schema, self.json, "Deployment template")
+            validation_success = self._validate_json_schema(template_schema, self.json, "Deployment template")
         except Exception as ex:  # Ignore other non shcema validation errors
             self.output.info("Unexpected error during deployment template schema validation, skip schema validation. Error:%s" % ex)
```

Comment:
exception happens and the validation is still success?

Reply:
Only schema failures should be treated as validation failure. Other exceptions such as network error should not fail the validation. iotedgedev will only print related error and skip validation.